src/libraries/System.Runtime.InteropServices/gen/Microsoft.Interop.SourceGeneration/BoundGenerators.cs

```diff
@@ -18,17 +18,29 @@ namespace Microsoft.Interop
 
     public class BoundGenerators
     {
-        public BoundGenerators(ImmutableArray<TypePositionInfo> elementTypeInfo, Func<TypePositionInfo, IMarshallingGenerator> generatorFactoryCallback)
+        public BoundGenerators(ImmutableArray<TypePositionInfo> elementTypeInfo, Func<TypePositionInfo, IMarshallingGenerator> generatorFactoryCallback, Action<TypePositionInfo, string> extendedInvariantViolationCallback)
         {
             ImmutableArray<BoundGenerator>.Builder allMarshallers = ImmutableArray.CreateBuilder<BoundGenerator>();
-            ImmutableArray<BoundGenerator>.Builder paramMarshallers = ImmutableArray.CreateBuilder<BoundGenerator>();
+            ImmutableArray<BoundGenerator>.Builder nativeParamMarshallers = ImmutableArray.CreateBuilder<BoundGenerator>();
+            ImmutableArray<BoundGenerator>.Builder managedParamMarshallers = ImmutableArray.CreateBuilder<BoundGenerator>();
             bool foundNativeRetMarshaller = false;
             bool foundManagedRetMarshaller = false;
+            TypePositionInfo? nativeExceptionInfo = null;
             NativeReturnMarshaller = new(new TypePositionInfo(SpecialTypeInfo.Void, NoMarshallingInfo.Instance), new Forwarder());
             ManagedReturnMarshaller = new(new TypePositionInfo(SpecialTypeInfo.Void, NoMarshallingInfo.Instance), new Forwarder());
+            ManagedExceptionMarshaller = new(new TypePositionInfo(SpecialTypeInfo.Void, NoMarshallingInfo.Instance), new Forwarder());
 
             foreach (TypePositionInfo argType in elementTypeInfo)
             {
+                if (argType.IsManagedExceptionPosition)
+                {
+                    Debug.Assert(nativeExceptionInfo is null);
```

Comment:
Is this going to be a user error or some internal logic issue? If user error I think we should fail hard.

Reply:
This would be a logic error in the usage of the APIs, not a user error.